libbeat/tests/docker/docker.go

```diff
@@ -65,6 +66,36 @@ func (c Client) ContainerStart(image string, cmd []string, labels map[string]str
 	return resp.ID, nil
 }
 
+// imagePull pulls an image
+func (c Client) imagePull(image string) (err error) {
+	ctx := context.Background()
+	_, _, err = c.cli.ImageInspectWithRaw(ctx, image)
+	if err == nil {
+		// Image already available, do nothing
+		return nil
+	}
+	for retry := 0; retry < 3; retry++ {
```

Comment:
Not sure if we could make it configurable like having this number of retries to be a function variable. Also maybe to add sleep time between each of the retries, however not sure if this is needed at all since not familiar with the nature of the errors that may occur here.

Suggestion would look like:
`imagePullWithRetry(image string, retries int, interval int)`

Reply:
Good suggestion, but I think this wouldn't be needed by now, and in any case we wouldn't expose these settings in the public methods.
I would prefer to  add them if we need it at some moment.

Reply:
👍